.github/CONTRIBUTING.md

```diff
@@ -232,19 +232,68 @@ We welcome any useful contribution! For your convenience here's a recommended wo
 
 3. **How to rebase my PR?**
 
-   We recommend creating a PR in separate branch other than `master`, especially if you plan submitting several changes and do not want to wait until the first one is resolved (we can work on them in parallel). Update your master with upstream (assuming you have already set [upstream](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/configuring-a-remote-for-a-fork))
+   We recommend creating a PR in separate branch other than `master`, especially if you plan submitting several changes and do not want to wait until the first one is resolved (we can work on them in parallel).
+   
+   First, make sure you have set [upstream](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/configuring-a-remote-for-a-fork) by running:
 
+   ```bash
+   git remote add upstream https://github.com/PyTorchLightning/pytorch-lightning.git
+   ```
+   
+   You'll know its set up right if you run `git remote -v` and see something similar to this:
+
+   ```bash
+   origin  https://github.com/{YOUR_USERNAME}/pytorch-lightning.git (fetch)
+   origin  https://github.com/{YOUR_USERNAME}/pytorch-lightning.git (push)
+   upstream        https://github.com/PyTorchLightning/pytorch-lightning.git (fetch)
+   upstream        https://github.com/PyTorchLightning/pytorch-lightning.git (push)
+   ```
+   
+   Now you can update your master with upstream's master by running:
    ```bash
    git fetch --all --prune
    git checkout master
    git merge upstream/master
    ```
 
-   checkout your feature branch
-
+   Finally, checkout your feature branch and rebase it with master before pushing up your feature branch:
    ```bash
    git checkout my-PR-branch
    git rebase master
    # follow git instructions to resolve conflicts
    git push -f
    ```
+
```

Comment:
```suggestion

Eventually, you can perform the rebasing directly from upstream after setting it up:
```bash
git fetch --all --prune
git rebase upstream/master
# follow git instructions to resolve conflicts
git push -f
```